tests/core/uc20-recovery/task.yaml

```diff
@@ -1,130 +1,168 @@
+        # TODO:UC20: there is only one system for now
+        jq .result.systems[0].current < systems.json | MATCH 'true'
+        label="$(jq -r .result.systems[0].label < systems.json)"
+        test -n "$label"
+        # make sure that the seed exists
+        test -d "/var/lib/snapd/seed/systems/$label"
+        jq -r .result.systems[0].actions[].mode < systems.json | sort | tr '\n' ' ' | MATCH 'install recover run'
+
+        # keep a copy of the systems dump for later reference
+        cp systems.json /writable/systems.json.run
+        echo "$label" > /writable/systems.label
+
+        # redirect shutdown command to our mock to observe calls and avoid racing
+        # with spread
+        mount -o bind "$PWD/mock-shutdown" /usr/sbin/shutdown
+
+        # request a recovery mode
+        echo "Request rebooting into recovery mode"
+        # response contains maintenance: {"maintenance":{"message":"system is restarting","kind":"system-restart"}}
+        test-snapd-curl.curl -X POST --data '{"action":"do","mode":"recover"}' \
+            -s --unix-socket /run/snapd.socket "http://localhost/v2/systems/$label" | \
+            MATCH '"system-restart"'
+
+        # snapd schedules a slow timeout and an immediate one, however it is
+        # scheduled asynchronously, try keep the check simple
+        # shellcheck disable=SC2016
+        retry -n 30 --wait 1 sh -c 'test "$(wc -l < /tmp/mock-shutdown.calls)" = "2"'
+        # an immediate reboot should have been scheduled
+        MATCH -- '-r \+0' < /tmp/mock-shutdown.calls
+
+        # restore shutdown so that spread can reboot the host
+        umount /usr/sbin/shutdown
+
+        # with the external backend, we do not have the special snapd snap with
+        # the first-boot run mode tweaks as created from $TESTLIB/prepare.sh's 
+        # repack_snapd_snap_with_deb_content_and_run_mode_firstboot_tweaks func
+        # so instead to get the spread gopath and other data needed to continue
+        # the test, we need to add a .ssh/rc script which copies all of the data
+        # from /host/ubuntu-data in recover mode to the tmpfs /home
+        if [ "$SPREAD_BACKEND" = "external" ]; then
+            mkdir -p /home/external/.ssh
+            chown external:external /home/external/.ssh
+            touch /home/external/.ssh/rc
+            chown external:external /home/external/.ssh/rc
+            cat <<-EOF > /home/external/.ssh/rc
+    #!/bin/sh
+    # added by spread tests
+    if grep -q 'snapd_recovery_mode=recover' /proc/cmdline; then
+        # use as sudo for proper permissions, assumes that the external user is
+        # a sudo user, which it must be by definition of it being the login user
+        # that spread uses
+        # silence the output so that nothing is output to a ssh command,
+        # potentially confusing spread
+        sudo mkdir /home/gopath 2> /dev/null > /dev/null
```

Comment:
According to sshd manpage, the script must not produce any output to stdout, but I'm not sure we should drop stderr here too. If those commands fail for any reason, it'll be hard to debug it.